src/Greenshot.Editor/Drawing/EmojiContainer.cs

```diff
@@ -0,0 +1,258 @@
+
+                _parent.Controls.Add(_emojiPickerHost);
+            }
+        }
+
+        private void HideEmojiPicker()
+        {
+            _emojiPicker?.ShowPopup(false);
+        }
+
+        protected override void OnDeserialized(StreamingContext streamingContext)
+        {
+            base.OnDeserialized(streamingContext);
+            Init();
+        }
+
+        private void Init()
+        {
+            CreateDefaultAdorners();
+
+            // Create WPF control that will be used to render the emoji
+            _image = new System.Windows.Controls.Image();
+            global::Emoji.Wpf.Image.SetSource(_image, Emoji);
+
+            PropertyChanged += OnPropertyChanged;
+        }
+
+        private void OnPropertyChanged(object sender, System.ComponentModel.PropertyChangedEventArgs e)
+        {
+            if (e.PropertyName.Equals(nameof(Selected)))
+            {
+                if (!Selected)
+                {
+                    HideEmojiPicker();
+                }
+                else if (Status == EditStatus.IDLE && _justCreated)
+                {
+                    // Show picker just after creation
+                    ShowEmojiPicker();
+                    _justCreated = false;
+                }
+            }
+        }
+
+        public override void Draw(Graphics graphics, RenderMode rm)
+        {
+            graphics.SmoothingMode = SmoothingMode.HighQuality;
+            graphics.InterpolationMode = InterpolationMode.HighQualityBicubic;
+            graphics.CompositingQuality = CompositingQuality.HighQuality;
+            graphics.PixelOffsetMode = PixelOffsetMode.HighQuality;
+
+            var rect = GuiRectangle.GetGuiRectangle(Left, Top, Width, Height);
+
+            var iconSize = Math.Min(rect.Width, rect.Height);
+
+            if (iconSize > 0)
+            {
+                _image.Measure(new Size(iconSize, iconSize));
+                _image.Arrange(new Rect(0, 0, iconSize, iconSize));
+
+                var renderTargetBitmap = new RenderTargetBitmap(iconSize, iconSize, 96, 96, PixelFormats.Pbgra32);
+                renderTargetBitmap.Render(_image);
+
+                using var bitmap = BitmapFromSource(renderTargetBitmap);
+
+                var rotationAngle = GetRotationAngle();
+                if (rotationAngle != 0)
+                {
+                    graphics.DrawImage(RotateImage(bitmap, rotationAngle), Bounds);
+                    return;
+                }
+
+                graphics.DrawImage(bitmap, Bounds);
+            }
+        }
+
+        private int GetRotationAngle()
+        {
+            var rotationAngle = 0;
+            if (Width < 0)
+            {
+                rotationAngle = Height > 0 ? 90 : 180;
+            }
+            else if (Height < 0)
+            {
+                rotationAngle = 270;
+            }
+
+            return rotationAngle;
+        }
+
+        private Bitmap BitmapFromSource(BitmapSource bitmapSource)
```

Comment:
Just a remark: It actually would make sense to provide this somewhere in Greenshot.Core, this feels very global.
Btw, I think it can be static... there is no access on instance variables so var I see.

Reply:
I wanted to keep my code in the same file but I can surely move it. Do you have a preference? `ImageHelpers.cs`?